src/org/freenetproject/freemail/wot/WoTConnection.java

```diff
@@ -32,7 +33,17 @@
 	 * @return all the OwnIdentities fetched from WoT
 	 * @throws PluginNotFoundException If the WoT plugin isn't loaded
 	 */
-	public List<OwnIdentity> getAllOwnIdentities() throws PluginNotFoundException;
+	List<OwnIdentity> getAllOwnIdentities() throws PluginNotFoundException, IOException, WoTException, InterruptedException;
```

Comment:
Does the exact exception matter at this level of interaction? If I want to retrieve all own identities (or set a property or whatever), do I really care about _why_ exactly it didn’t work? Is there anything I as caller of that method can do to change the outcome? I don’t think so, so can you condense the different exceptions into fewer, optimally a single one?